make/autoconf/spec.gmk.template

```diff
@@ -169,6 +169,9 @@ endif
 SYSROOT_CFLAGS := @SYSROOT_CFLAGS@
 SYSROOT_LDFLAGS := @SYSROOT_LDFLAGS@
 
+# dev or ci
+BUILD_ENV := @BUILD_ENV@
```

Comment:
I think the `BUILD_ENV` could be use in command line also. So, should we update the doc/building.md sync with this change.

Reply:
What do you mean? This is controlled by the `--with-build-env=[ci|dev]` option. It just needed to be exported from the configure script to the make files.

I am not sure if `--with-build-env` is documented in building.md, but in any case we just list a short selection of options there; the documentation points to `configure --help` to list all currently available options. (It is not feasible to keep the documentation in sync for every possible option.)

Reply:
At first I mistakenly thought that the value of the BUILD_ENV variable was read from the environment variable